edgelet/hsm-rs/src/crypto.rs

```diff
@@ -194,6 +194,30 @@ fn make_certification_props(props: &CertificateProperties) -> Result<CERT_PROPS_
             unsafe { cert_properties_destroy(handle) };
             ErrorKind::CertProps
         })?;
+
+    if props.san_entries.len() > 0 {
+        let result: Result<Vec<CString>, NulError> = props
+            .san_entries
+            .iter()
+            .map(|s: &String| CString::new(s.clone()))
+            .collect();
```

Comment:
Not sure I am reading this right. Isn't `map` returning an `Iterator` of `Result<CString, NulError>`? If so, wouldn't the `collect` call then produce a `Vec<Result<CString, NulError>>` instead of a `Result<Vec<CString>, NulError>`?

Reply:
collect() implements trait FromIterator https://doc.rust-lang.org/std/iter/trait.FromIterator.html.
Specifically,
impl<A, E, V> FromIterator<Result<A, E>> for Result<V, E> where    V: FromIterator<A>,